### ISSUE TITLE
Quantifier reported as useless, but it isn't

### ISSUE THREAD

**Issue by Slapbox** [ota-meshi/eslint-plugin-regexp#310]
**Information:**
- ESLint version: 7.32.0
- `eslint-plugin-regexp` version: 1.0.0

**Description**

no-useless-quantifier reports the first `?` in RegExp below as useless, but code behavior changes if removed.

```js
const test = /(\w*)?.*?/
```

**Comment by Slapbox** [comment]
This might be a lie, because reverting that didn't fix our issue but `git bisect` leads us right here and it's seemingly the only relevant change so I haven't a clue.

**Comment by RunDevelopment** [comment]
I don't know what your bug is but `/(\w*)?.*?/` should behave exactly the same as `/(\w*).*?/` (and even `/(\w*)/`).

**Comment by Slapbox** [comment]
Yep, sorry for the noise.
